core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2632,11 +2632,18 @@ private[spark] object Utils extends Logging {
     // arbitrary property contained the term 'password', we may redact the value from the UI and
     // logs. In order to work around it, user would have to make the spark.redaction.regex property
     // more specific.
-    kvs.map { case (key, value) =>
-      redactionPattern.findFirstIn(key)
-        .orElse(redactionPattern.findFirstIn(value))
-        .map { _ => (key, REDACTION_REPLACEMENT_TEXT) }
-        .getOrElse((key, value))
+    kvs.map {
+      case (key: String, value: String) =>
+        redactionPattern.findFirstIn(key)
+          .orElse(redactionPattern.findFirstIn(value))
+          .map { _ => (key.asInstanceOf[K], REDACTION_REPLACEMENT_TEXT.asInstanceOf[V]) }
+          .getOrElse((key.asInstanceOf[K], value.asInstanceOf[V]))
+      case (key, value: String) =>
+        redactionPattern.findFirstIn(value)
+          .map { _ => (key, REDACTION_REPLACEMENT_TEXT.asInstanceOf[V]) }
+          .getOrElse((key, value.asInstanceOf[V]))
+      case (key, value) =>
+        (key, value)
```

Comment:
Ur, it seems to be too verbose. For readability, can we use `asInstanceOf` once instead of 6 times?
```scala
    kvs.map {
      case (key: String, value: String) =>
        redactionPattern.findFirstIn(key)
          .orElse(redactionPattern.findFirstIn(value))
          .map { _ => (key, REDACTION_REPLACEMENT_TEXT) }
          .getOrElse((key, value))
      case (key, value: String) =>
        redactionPattern.findFirstIn(value)
          .map { _ => (key, REDACTION_REPLACEMENT_TEXT) }
          .getOrElse((key, value))
      case (key, value) =>
        (key, value)
    }.asInstanceOf[Seq[(K, V)]]
```

Reply:
done

Reply:
Just out of curiosity, why not to 1 as I gave?

Reply:
Not sure about the performance impact of casting the entire collection. Do you have more insight?

Reply:
It's `map` operation. This already casts the all items in the collection, doesn't it?

Reply:
Don't think map function casts anything implicitly.
My perf concern is probably overblown.